src/jit/gentree.cpp

```diff
@@ -7462,9 +7477,6 @@ void Compiler::gtBlockOpInit(GenTreePtr result, GenTreePtr dst, GenTreePtr srcOr
     result->gtFlags |= dst->gtFlags & GTF_ALL_EFFECT;
     result->gtFlags |= result->gtOp.gtOp2->gtFlags & GTF_ALL_EFFECT;
 
-    // TODO-1stClassStructs: This should be done only if the destination is non-local.
-    result->gtFlags |= (GTF_GLOB_REF | GTF_ASG);
```

Comment:
No, the destination will have GTF_GLOB_REF if necessary (and will be propagated above), and the assignment will already have GTF_ASG set.

---
In reply to: [84718619](https://github.com/dotnet/coreclr/pull/7677#discussion_r84718619) [](ancestors = 84718619)